tests/test_opentracing_hook.py

```diff
@@ -0,0 +1,351 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import threading
+import time
+import traceback
+import urllib
+
+import mock
+import opentracing
+import pytest
+import tornado
+import tornado.gen
+import tornado.web
+from basictracer import BasicTracer, SpanRecorder
+from tchannel.context import RequestContextProvider
+from tchannel.tracing.opentracing_hook import OpenTracingHook, SpanWrapper
+from tchannel import TChannel, Response
+from tornado.httpclient import HTTPRequest
+from tornado.stack_context import StackContext
+from opentracing import Format
+from opentracing_instrumentation.request_context import \
+    RequestContext, RequestContextManager, \
+    span_in_stack_context, get_current_span
+from opentracing_instrumentation.client_hooks.tornado_http import \
+    install_patches, reset_patchers
+
+
+BAGGAGE_KEY = b'baggage'
+
+
+@pytest.fixture
+def span_recorder():
+    class InMemoryRecorder(SpanRecorder):
+        def __init__(self):
+            self.spans = []
+            self.mux = threading.Lock()
+
+        def record_span(self, span):
+            with self.mux:
+                self.spans.append(span)
+
+        def get_spans(self):
+            with self.mux:
+                return self.spans[:]
+
+    return InMemoryRecorder()
+
+
+# noinspection PyShadowingNames
+@pytest.fixture
+def tracer(span_recorder):
+    return BasicTracer(recorder=span_recorder)
+
+
+class MockRequestContext(RequestContext):
+    """
+    This class extends the default `opentracing_instrumentation.RequestContext`
+    with a `parent_tracing` attribute, making it suitable for both TChannel
+    and `opentracing_instrumentation`.
+    """
+    __slots__ = ('parent_tracing', )
+
+    def __init__(self, span, parent_tracing=None):
+        super(MockRequestContext, self).__init__(span=span)
+        self.parent_tracing = parent_tracing
+
+
+class OpenTracingRequestContextProvider(RequestContextProvider):
+    """
+    Extends TChannel's RequestContextProvider by substituting the internal
+    `RequestContext` with `opentracing_instrumentation.RequestContext`.
+    """
+
+    def get_current_context(self):
+        """Returns a Context that has parent_tracing attribute"""
+        context = RequestContextManager.current_context()
+        if not context:
+            return None
+        if not hasattr(context, 'parent_tracing'):
+            context = MockRequestContext(
+                span=context.span,
+                parent_tracing=SpanWrapper(span=context.span)
```

Comment:
@abhinav not sure what you mean. The primary (opentracing) context does not have a `parent_tracing` attribute, but tchannel expects it from `get_current_context` call, so I replace the opentracing context with a tchannel-compatible extension.